mora/service/cpr.py

```diff
@@ -48,12 +48,20 @@ def search_cpr():
     """
     cpr = flask.request.args['q']
 
-    sp_data = get_citizen(cpr)
-
-    if sp_data:
-        return flask.jsonify(format_cpr_response(sp_data, cpr))
-    else:
-        return '', 404
+    try:
+        sp_data = get_citizen(cpr)
+    except KeyError:
+        raise exceptions.NotFoundError(
+            'no such person found',
+            cpr=cpr,
+        )
+    except ValueError:
```

Comment:
Why catch and re-raise these exceptions? Wouldn't it make more sense to just raise the relevant errors in `get_citizen` directly?

Reply:
Layering? Figured the `integrations` package wasn't as MO-specific, and should raise plain exceptions, leaving the interpretation to the `cpr_lookup` service.

Reply:
I did intend it as something MO-specific, actually. The `integrations` package is to contain all of the adapters and logic for connecting to external systems - but I see your point. This way we could potentially extract this and reuse it somewhere else.

Reply:
Anyway, my thinking was that `KeyError` and `ValueError` talk of arguments to the current function, whereas the `exceptions` module is for shortcutting evaluation with an HTTP error — and that felt like something an integration shouldn't do.